docs/src/man/comparisons.md

```diff
@@ -1,77 +1,173 @@
 # Comparisons
 
-This section compares DataFrames.jl with other data manipulation frameworks.
+This section compares DataFrames.jl with other data manipulation frameworks in Python, R, and Stata.
+
+A sample data set can be created using the following code:
+
+```julia
+df = DataFrame(id = 'a':'f', grp = repeat(1:2, 3), x = 6:-1:1, y = 4:9, z = [3:7; missing])
+```
+
+## Comparison with the Python package pandas
+
+The following table compares the main functions of DataFrames.jl with the Python package pandas (version 1.1.0):
+
+```python
+df = pandas.DataFrame({'grp': [1, 2, 1, 2, 1, 2],
+                       'x': range(6, 0, -1),
+                       'y': range(4, 10),
+                       'z': [3, 4, 5, 6, 7, None]},
+                      index = list('abcdef'))
+```
+
+By comparison, this pandas data frame has `a` to `f` as row indices rather than a separate `id` column.
+
+### Accessing data
+
+| Operation                  | pandas                 | DataFrames.jl            |
+|:---------------------------|:-----------------------|:-------------------------|
+| Cell indexing by location  | `df.iloc[1, 1]`        | `df[2, 2]`               |
+| Row slicing by location    | `df.iloc[1:3]`         | `df[2:3, :]`             |
+| Column slicing by location | `df.iloc[:, 1:]`       | `df[:, 2:end]`           |
+| Row indexing by label      | `df.loc['c']`          | `df[df.id .== 'c', :]`   |
+| Column indexing by label   | `df.loc[:, 'x']`       | `df[:, :x]`              |
+| Column slicing by label    | `df.loc[:, ['x','z']]` | `df[:, [:x, :z]]`        |
+|                            | `df.loc[:, 'x':'z']`   | `df[:, Between(:x, :z)]` |
+| Mixed indexing             | `df.loc['c'][1]`       | `df[df.id .== 'c', 2]`   |
+
+Note that Julia uses 1-based indexing, inclusive on both ends. A special keyword `end` can be used to
+indicate the last index.
+
+### Common operations
+
+| Operation                | pandas                                                | DataFrames.jl                                                 |
+|:-------------------------|:------------------------------------------------------|:--------------------------------------------------------------|
+| Reduce multiple values   | `df['z'].mean(skipna = False)`                        | `combine(df, :z => mean)`                                     |
+|                          | `df['z'].mean()`                                      | `combine(df, :z => mean ∘ skipmissing => :z_mean)`            |
```

Comment:
The python code is not the same. It is just `mean(df.z)` and `mean(skipmissing(df.z))`. What would be the same result as in Julia, i.e. `DataFrame` with column `z_mean`?

Reply:
That's a good question.  Python auto-collapses dimensions. When you aggregate several columns, it becomes a series with row index equals to the column name. When you aggregate a single column, it gives you a single number.

```python
>>> df['z'].mean(skipna = False)
nan
>>> df['z'].mean()
5.0
>>> df[['x', 'z']].mean()
x    3.5
z    5.0
```

Maybe we should have separate examples?
```julia
julia> mean(df.z)
missing

julia> mean(skipmissing(df.z))
5.0

julia> combine(df, [:x, :z] .=> mean)
1×2 DataFrame
│ Row │ x_mean  │ z_mean  │
│     │ Float64 │ Missing │
├─────┼─────────┼─────────┤
│ 1   │ 3.5     │ missing │
```
```julia
```

Reply:
I am OK with choosing the examples you feel are most instructive given:
* the pandas and DataFrames.jl codes provide the same results (in terms of type and shape)
* we use DataFrames.jl functionality (i.e. I feel that showing `mean(df.z)` is unrelated to DataFrames.jl - it is a functionality of Statistics.jl)

Reply:
And what does `df['z'].mean()` produce in pandas?

Reply:
pandas returns a single numeric value.  See examples above.

@bkamins I'm torn because I hope this page can give simple direction to pandas users for common use cases. If we stick to DataFrames-only verbs then we cannot give them the most logical way to solve the same problem. Pandas doesn't work the same way and does not map 1-to-1 to DataFrames.  Having said that, it seems reasonable to me to include something like `mean(df.z)`...

Reply:
> pandas returns a single numeric value. See examples above.

Even with the double `[`?

Reply:
>  it seems reasonable to me to include something like `mean(df.z)`

OK - let us have these, but maybe in a section like "things you can do just using Julia Base and standard library" (probably we need a better name) 😄, so that users are clear that they are not using the functionality of DataFrames.jl.

Reply:
As a side note - so it is impossible to use a "piping" style in pandas like in e.g. dplyr? @pdeffebach will probably find it intrtesting 😄.

Reply:
You can! you just stack the `.` calls over and over again as far as I understand. Perhaps it could be added.

Reply:
I understand you are not - if at some point you stop being a data frame you cannot go back to be a data frame via chaining `.` because e.g. a vector does not have a method to make a data frame out of it.

Reply:
@nalimilan it's slightly different with double `[`. It returns a series with the column name as row index:

```
>>> df['z'].mean()
5.0
>>> df[['z']].mean()
z    5.0
dtype: float64
```

Reply:
OK. I like this idea and have update the pandas example. See latest commit.